src/CTA.WebForms/Helpers/TagConversion/TagConfigParser.cs

```diff
@@ -49,19 +74,68 @@ public TagConfigParser(string configsDir)
                     // block before we attempt to add the converter to the results dictionary
                     converter.Validate();
 
-                    if (!result.TryAdd(converter.TagName, converter))
+                    if (!TagConverterMap.TryAdd(converter.TagName, converter))
                     {
                         LogHelper.LogError($"{Rules.Config.Constants.WebFormsErrorTag}Failed to add valid converter of type " +
                             $"{converter?.GetType().Name} for {converter?.TagName} to concurrent dictionary for config at {filePath}");
                     }
+
+                    // Even if adding to the dictionary fails, we want to return the validated converter,
+                    // the dictionary only serves to improve performance when retrieving converters
+                    return converter;
```

Comment:
The reason we use try add here is because its a concurrent dictionary. We need to first check if there is an existing key in the dictionary, then attempt to add it, but doing this is non-atomic. The concurrent dictionary's TryAdd basically just does the extra error handling if there was a change to the dictionary between the ContainsKey call and the accessor call that creates an invalid state. Will modify the error message to be more specific

Reply:
Got it, that makes sense. Ty for the clarification